core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -135,13 +145,20 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     private val otherFields = mutable.WeakHashMap[String, StringField]() // weak so that it is GCed when unused
 
     def addField(name: String, value: String): Unit = {
+      // Use PartKeyIndexBenchmark to measure indexing performance before changing this
+      if (facetEnabled && name.nonEmpty && value.nonEmpty) {
+        facetsConfig.setRequireDimensionDrillDown(name, false)
+        facetsConfig.setIndexFieldName(name, FACET_FIELD_PREFIX + name)
+        document.add(new SortedSetDocValuesFacetField(name, value))
```

Comment:
From ``SortedSetDocValuesFacetField`` the documentation, Faceting is 25% slower and more importantly there is an additional cost for ``IndexReader`` open at the cost of lower RAM usage. Something to keep in mind.

Reply:
As we discovered, that javadoc is from older lucene version. Latest 8.x does not have that in the javadoc. Things have probably changed.